### ISSUE TITLE
Add support for Pre Tie 95 games

### ISSUE THREAD

**Issue by Danfun64** [MikeG621/YOGEME#14]
I know this is called "Ye Olde GALACTIC EMPIRE Mission Editor", but I'd really appreciate it if you supported every version of X-Wing (DOS Floppy, DOS Floppy with Expansions, DOS CD, Mac CD, Win95) as well as the versions of Tie Fighter you currently don't support (DOS Floppy, DOS Floppy with Expansions, DOS CD, Mac CD)

This might be one of the only open source editors for the X-Wing series, and i'd love it if it could edit all of them.

**Comment by MikeG621** [comment]
The first problem with supporting everything is that I need a copy of
everything. The second is that I would likely need to build a new "old"
computer to run the oldest variants since modern computers have a hard time
or can't run the oldest versions.

Adding X-wing is technically possible (pretty sure I have the Steam
version) although would be very time consuming as I'd have to research the
file formats first. So if I were to start on that it won't be done any time
soon. Quite frankly I do not have the time available to do this without it
taking years.

As far as the other versions of TIE, I *believe* the mission file format
remained the same, but I would also need additional information to fully
activate other features such as platform detection, backdrop dialog, test,
etc.  Those mostly rely on the appropriate settings in the options dialog
being set (installation directory) which can be set manually if it's not
detected, as well as various file names used by TIE that may be
version-specific. So it's entirely possible that missions used for the
other versions of TIE work perfectly fine, just needs an extra step or two
on the user's part to get there and YOGEME might not have access to
everything. But if I can dig up that information it may be possible to add
those.

**Comment by Danfun64** [comment]
You don't need to build a "new 'old' computer" to run the oldest variants. Mac ports aside, almost every version (except X-Wing and Tie Fighter Floppies without Expansions) are available on GOG and Steam. 

An important thing to note is that the briefing screen in the Mac and Windows versions of X-Wing is bigger and shows more text and an unzoomed map, as well as the ability to show more text in flight because of the 640x480 cockpit. Likewise the cockpit in Tie Fighter DOS CD, Mac, and Windows probably allow for more text because of the choice of being in 640x480. However in all three of those versions (despite the enhancements to the menus of the latter two) the briefing map and text appears to be exactly the same in size. 

Also, for some reason the Windows (and I'm assuming the Mac) versions of X-Wing have two missions.ldf files, the original in the resources folder and a "high res" version in the missions folder. This resulted in the breaking of the X-Wing 2 mod which runs in both DOS versions.

So basically you have custom missions that ATM only run on the DOS ports, and on the other you have custom missions that technically _can_ run on the DOS ports but only display properly on the Windows and Mac ports. Sound confusing?

**Comment by MikeG621** [comment]
Now that you mention it, you're right about the different versions on
Steam, I forgot they added that extra launcher to switch between versions
(I think I'm bypassing it since I always ran TIE95). I do know that TIE95
switched resolutions, that was primarily due to using the flight engine
from XvT. The different briefing screen might be interesting since it'd
have to be scaled differently. The briefing officers dialog would also have
to use different graphics.

I'll poke around on the TIE side of things

On Nov 4, 2017 15:02, "Danfun64" <notifications@github.com> wrote:

You don't need to build a "new 'old' computer" to run the oldest variants.
Mac ports aside, almost every version (except X-Wing and Tie Fighter
Floppies without Expansions) are available on GOG and Steam.

An important thing to note is that the briefing screen in the Mac and
Windows versions of X-Wing is bigger and shows more text and an unzoomed
map, as well as the ability to show more text in flight because of the
640x480 cockpit. Likewise the cockpit in Tie Fighter DOS CD, Mac, and
Windows probably allow for more text because of the choice of being in
640x480. However in all three of those versions (despite the enhancements
to the menus of the latter two) the briefing map and text appears to be
exactly the same in size.

Also, for some reason the Windows (and I'm assuming the Mac) versions of
X-Wing have two missions.ldf files, the original in the resources folder
and a "high res" version in the missions folder. This resulted in the
breaking of the X-Wing 2 mod which runs in both DOS versions.

So basically you have custom missions that ATM only run on the DOS ports,
and on the other you have custom missions that technically *can* run on the
DOS ports but only display properly on the Windows and Mac ports. Sound
confusing?

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/MikeG621/YOGEME/issues/14#issuecomment-341921754>, or mute
the thread

**Comment by Danfun64** [comment]
I'm pretty sure the interface for the imperial officer is the same on both
dos and Windows/Mac

**Comment by MikeG621** [comment]
Xwing95 support has been implemented

**Comment by MikeG621** [comment]
Wireframes support 95 OPTs older DOS/CD versions

**Comment by RedDragonWebDesign** [comment]
I'm a bit late to the conversation. But [DOSBox](https://www.dosbox.com/) can be a great resource for getting old games running without having to build a DOS computer.
